integration_tests/src/lib.rs

```diff
@@ -1764,9 +1852,172 @@ mod test {
         Ok(())
     }
 
-    // The following tests work at the "guest" layer.  The volume
+    #[tokio::test(flavor = "multi_thread", worker_threads = 8)]
+    async fn integration_test_snapshot_backed_vol() -> Result<()> {
+        // Test using a "snapshot" (for this test, downstairs booted read-only)
+        // as a read-only parent.
+
+        const BLOCK_SIZE: usize = 512;
+
+        // boot three downstairs, write some data to them, then change to
+        // read-only.
+        let mut test_downstairs_set =
+            TestDownstairsSet::new(54082, 54083, 54084, false)?;
+
+        let mut volume = Volume::new(BLOCK_SIZE as u64);
+        volume.add_subvolume_create_guest(
+            test_downstairs_set.opts(),
+            0,
+            None,
+        )?;
+
+        volume.activate(0)?;
+
+        let random_buffer = {
+            let mut random_buffer = vec![0u8; volume.total_size()? as usize];
+            rand::thread_rng().fill(&mut random_buffer[..]);
+            random_buffer
+        };
+
+        volume
+            .write(
+                Block::new(0, BLOCK_SIZE.trailing_zeros()),
+                Bytes::from(random_buffer.clone()),
+            )?
+            .block_wait()?;
+
+        volume.deactivate()?.block_wait()?;
```

Comment:
Asking primarily for my own benefit - does this need a flush before the deactivation, or does the deactivation send a flush implicitly?

Reply:
It's a good question - the deactivation sends a final flush. Check out the relevant function is [`set_deactivate`](https://github.com/oxidecomputer/crucible/blob/5263f0a147185e63cf64ea7ae05837ef1b501a63/upstairs/src/lib.rs#L3698):

```
     * For a deactivation to complete, we need to:
     * Stop all incoming IO.
     * Let any outstanding IO finish.
     * Submit a final flush to all downstairs.
     * Wait for that final flush to finish (on all downstairs).
```